src/exchanges/BitfinexClient.ts

```diff
@@ -152,6 +209,42 @@ export class BitfinexClient extends BasicClient {
         }
     }
 
+    /**
+     * Handle heartbeat messages on each channel.
+     */
+    protected _onHeartbeatMessage(msg: any, channel: any) {
```

Comment:
this will handle each channel's heartbeat event by calling the corresponding heartbeat handler. If `enableEmptyHeartbeatEvents` is true each of these will fire an empty event for that channel with the `sequenceId` included